### DISCUSSION TITLE
Virtual Scrolling not work with DispatchFunc

### DISCUSSION THREAD

**Discussion by tretyakdnua** [komarovalexander/ka-table#436] [General]
Hi, Alexander!
It seems that when virtualScrolling is enabled we don't have opportunity to work with dispatch={dispatch} function. It breaks table layout and only first couple rows are visible. Could you tell, please, why it work's that way. It becomes very tricky to find workarounds instead to use actions.

**Comment by komarovalexander** [comment]
Hi @tretyakdnua can you reproduce that on stackblitz? https://stackblitz.com/edit/table-many-rows-ts?file=Demo.tsx

**Reply by tretyakdnua** [reply]
Sorry, but I don't know how to save and share edited code to you in stackblitz. But I do edits and I can see the bug. Just try it by yourself, you just need to add to <Table dispatch={dispatch} ... and add a function as in docs like this: const dispatch: DispatchFunc = (action) => {}
and that's it. You can see that only first 20 rows are visible!

**Reply by tretyakdnua** [reply]
![vs](https://github.com/user-attachments/assets/7d7fc54d-8f69-42c1-87a3-20983cf07421)

**Reply by komarovalexander** [reply]
@tretyakdnua what is the purpose to use dispatch in your example? what do you truing to achieve? do you need to subscribe to some action or something else?

this approach that you show on image is controlled approach and here is an example how it should work https://stackblitz.com/edit/table-many-rows-js?file=Demo.js 

but looks like in your case you just need to subscribe to dispatched actions that way you can use useTable({ onDispatch: () => {} }), example https://stackblitz.com/edit/table-many-rows-ts-r4phkm?file=Demo.tsx

**Reply by tretyakdnua** [reply]
Thank's! Provided by you controlled approach actually produce performance rendering leaks on FF with big delays.Yes, I need to subscribe to events.

**Reply by tretyakdnua** [reply]
But approach with useTable works great on FF and Chrome. Thx!
